src/styles/main.scss

```diff
@@ -0,0 +1,81 @@
+$day-bg-color: #eee;
+$day-size: 100px;
+$gap-size: 1px;
+$font-size: 30px;
+$font-color: #000;
+$hover-color: #ffbfcb;
+$widthCalendar: 7 * $day-size + $gap-size * 6;
+
+.body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  height: 100vh;
+  align-items: center;
+}
+
+.calendar {
+  width: $widthCalendar;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+
+  $days: (
+    mon: 0,
+    tue: 1,
+    wed: 2,
+    thu: 3,
+    fri: 4,
+    sat: 5,
+    sun: 6,
+  );
+
+  @each $day, $i in $days {
+    &--start-day-#{$day} {
+      .calendar__day:nth-child(1) {
+        margin-left: ($day-size * ($i) + ($day-size * ($i) / 100));
+      }
+    }
+  }
+
+  @for $a from 28 through 31 {
+    &--month-length-#{$a} {
+      @for $i from $a through 31 {
+        .calendar__day:nth-child(#{$i + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    background-color: $day-bg-color;
+    border: solid #000 1px;
+    align-content: center;
+    text-align: center;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: $font-size;
+    transition: 0.5s;
+
+    &:hover {
+      background-color: $hover-color;
+      transform: translate(0, -20px);
+      cursor: pointer;
+    }
+
+    @for $i from 1 through 31 {
+      .calendar__day:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}):before {
+        content: '#{$i}';
+      }
+    }
```

Comment:
There is a redundancy here. The `@for` loop from lines 69 to 73 and the one from lines 74 to 79 are doing the same thing. You can remove one of them to avoid redundancy.